packages/trader/src/Modules/Trading/Components/Form/ContractType/contract-type-widget.tsx

```diff
@@ -80,79 +84,93 @@ const ContractTypeWidget = ({ is_equal, name, value, list, onChange, languageCha
-        const multipliers_category = ordered_list.filter(({ label }) => label === localize('Multipliers'));
-        const options_category = ordered_list.filter(
+        list?.sort((a, b) => order_arr.indexOf(a.key) - order_arr.indexOf(b.key));
+        const accumulators_category = list?.filter(({ label }) => label === localize('Accumulators'));
+        const multipliers_category = list?.filter(({ label }) => label === localize('Multipliers'));
+        const options_category = list?.filter(
             ({ label }) => label !== localize('Multipliers') && label !== localize('Accumulators')
         );
 
-        const categories = [];
+        const categories: TList[] = [];
 
-        if (list.length > 0) {
+        if (list && list.length > 0) {
             categories.push({
                 label: localize('All'),
-                contract_categories: [...ordered_list],
+                contract_categories: [...list],
                 key: 'All',
+                icon: '',
             });
         }
 
-        if (multipliers_category.length > 0) {
+        if (multipliers_category && multipliers_category.length > 0) {
             categories.push({
                 label: localize('Multipliers'),
                 contract_categories: multipliers_category,
                 key: 'Multipliers',
+                icon: '',
             });
         }
 
-        if (options_category.length > 0) {
+        if (options_category && options_category.length > 0) {
             categories.push({
                 label: localize('Options'),
                 contract_categories: options_category,
+                // @ts-expect-error The type of the component prop in VerticalTab.Headers is wrong and it should be JSX.Element
                 component: options_category.some(category => category.key === 'Vanillas') && (
                     <span className='dc-vertical-tab__header--new'>{localize('NEW')}!</span>
                 ),
                 key: 'Options',
             });
         }
 
-        if (accumulators_category.length > 0) {
+        if (accumulators_category && accumulators_category.length > 0) {
             categories.push({
                 label: localize('Accumulators'),
                 contract_categories: accumulators_category,
+                // @ts-expect-error The type of the component prop in VerticalTab.Headers is wrong and it should be JSX.Element
                 component: <span className='dc-vertical-tab__header--new'>{localize('NEW')}!</span>,
                 key: 'Accumulators',
             });
         }
 
         return categories.map(contract_category => {
-            contract_category.contract_types = contract_category.contract_categories.reduce(
-                (aray, x) => [...aray, ...x.contract_types],
+            const contract_types = contract_category.contract_categories.reduce<TContractType[]>(
```

Comment:
@henry-deriv great job! it's a nice way to add a type to reduce() which I almost forgot about!